spec/21-http_response_header_format.md

```diff
@@ -1,113 +1,107 @@
-# Trace Context HTTP Response Headers Format
+# Trace Context Server Timing Metric Format
 
-This section describes the binding of the distributed trace context to the `traceresponse` HTTP header.
+This section describes the binding of the distributed trace context to a metric in the Server Timing HTTP header.
 
-## Traceresponse Header
+## Trace Context Metric
 
-The `traceresponse` HTTP response header field identifies a completed request in a tracing system. It has four fields:
+The trace context metric identifies a completed request in a tracing system. It has four params:
 
-* `version`
-* `trace-id`
-* `child-id`
-* `trace-flags`
+* `tid` - required
+* `cid` - required
+* `v` - optional
+* `flags` - optional
 
-### Header Name
+Example server timing header with trace context metric:
 
-Header name: `traceresponse`
-
-The header name is [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive). That is, `TRACERESPONSE`, `TraceResponse`, and `traceresponse` are considered the same header. The header name is a single word; it does not contain any delimiters such as a hyphen.
+```
+server-timing: trace;tid=0af7651916cd43dd8448eb211c80319c,cid=b7ad6b7169203331
+```
```

Comment:
With respect to making the trace context available to client side JS, does that really work if we use arbitrary key value pairs like that? The way I used `Server-Timing` in the past was to always strictly stick with the `desc` and `dur` attributes from the [PerformanceServerTiming Interface](https://www.w3.org/TR/server-timing/#the-performanceservertiming-interface).

See also:
* https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing#performanceservertiming_interface

I would have expected that we were going to define _one_ metric (`trace`, `span`, `tracecontext`, `traceresponse`, which ever name we settle on) and put all the information we need into the `desc` field of that metric. That way, I believe it would be correctly exposed by browsers via the performance API. (Side note: We probably do not have a use for the `dur` field, so we would just omit it.)

For reference, that is also what OpenTelemetry is discussing: https://github.com/open-telemetry/opentelemetry-specification/issues/3811:

```
Server-Timing: traceparent;desc="00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01"
```

It is basically the previous `traceresponse` format, in the `desc` field of a `Server-Timing` metric. 

Of course, I might be missing something here. If it is possible to transport arbitrary key-value pairs like suggested in this PR and have them exposed to client side JS via the ServerTiming API (and browser support for that is on par with the `desc` attribute approach), the suggestion in this PR is cleaner.

Reply:
For reference, here is a quick test app I used:

```
const http = require('node:http');
const hostname = '127.0.0.1';
const port = 3000;

const server = http.createServer((req, res) => {
  res.statusCode = 200;
  res.setHeader('Content-Type', 'text/html');
  res.setHeader('Server-Timing', 'trace;tid=0af7651916cd43dd8448eb211c80319c,cid=b7ad6b7169203331');
  res.end(`
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Server-Timing Test Page</title>
  <script>
    const metrics = performance.getEntriesByType('navigation')[0]['serverTiming'];
    console.log(JSON.stringify(metrics, null, 2));
  </script>
</head>
<body>
Ohai!
</body>
</html>
`);
});

server.listen(port, hostname, () => {
  console.log(`Server running at http://${hostname}:${port}/`);
});
```

This results in the following output in the browser's console (in Chrome):

```
[
  {
    "name": "trace",
    "duration": 0,
    "description": ""
  },
  {
    "name": "cid",
    "duration": 0,
    "description": ""
  }
]
```

So `trace` becomes a metric without any values, `cid` becomes a metric without any values and `tid` gets discarded.

Changing the setHeader line to `res.setHeader('Server-Timing', 'traceresponse;desc="00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-03"')` yields the following:

```
[
  {
    "name": "traceresponse",
    "duration": 0,
    "description": "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-03"
  }
]
```

Reply:
I think you're right. I'll update the PR

Reply:
Done in ca90939974154fbe77a80ad2c1316da3b1ea11b2